src/GraphQL/SortBlockMutationCreator.php

```diff
@@ -0,0 +1,93 @@
+<?php
+namespace DNADesign\Elemental\GraphQL;
+
+use DNADesign\Elemental\Models\BaseElement;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\Type;
+use SilverStripe\GraphQL\MutationCreator;
+use SilverStripe\GraphQL\OperationResolver;
+use SilverStripe\ORM\DB;
+
+class SortBlockMutationCreator extends MutationCreator implements OperationResolver
+{
+    public function attributes()
+    {
+        return [
+            'name' => 'sortBlock',
+            'description' => 'Changes the sort position of an element'
+        ];
+    }
+
+    public function type()
+    {
+        return $this->manager->getType('Block');
+    }
+
+    public function args()
+    {
+        return [
+            'ID' => ['type' => Type::nonNull(Type::id())],
+            'AfterBlockID' => ['type' => Type::nonNull(Type::id())],
+        ];
+    }
+
+    public function resolve($object, array $args, $context, ResolveInfo $info)
+    {
+        $block = BaseElement::get_by_id($args['ID']);
+
+        if (!$block) {
+            throw new \InvalidArgumentException(sprintf(
+                '%s# not found',
+                ElementArea::class,
+                $args['ID']
+            ));
+        }
+
+        $parentId = $block->ParentID;
+        $blockPosition = $block->Sort;
+
+        $sortAfterPosition = 0;
+        $afterBlockID = $args['AfterBlockID'];
+        if ($afterBlockID) {
+            $afterBlock = BaseElement::get_by_id($afterBlockID);
+
+            if (!$afterBlock) {
+                throw new \InvalidArgumentException(sprintf(
+                    '%s#%s not found',
+                    BaseElement::class,
+                    $afterBlockID
+                ));
+            }
+            if ($afterBlock->ParentID !== $parentId) {
+                throw new \InvalidArgumentException(sprintf(
+                    '%s# not found',
+                    BaseElement::class,
+                    $parentId
+                ));
+            }
+
+            $sortAfterPosition = $afterBlock->Sort;
+        }
+
+        if ($sortAfterPosition < $blockPosition) {
+            $operator = '+';
+            $filter = "Sort > $sortAfterPosition && Sort < $blockPosition";
+            $newBlockPosition = $sortAfterPosition + 1;
+        } else {
+            $operator = '-';
+            $filter = "Sort <= $sortAfterPosition && Sort > $blockPosition";
+            $newBlockPosition = $sortAfterPosition;
+        }
+
+        DB::query("
+            UPDATE Element
+            SET Sort = (Sort $operator 1)
+            WHERE $filter && ParentId = $parentId 
+        ");
```

Comment:
I don't agree with this approach for two reasons:

* We should be avoiding direct SQL queries, and at worst should be using SQLUpdate with quoted table and field names so it will work with DB engines other than MySQL
* Updating the sort order of other elements in the same parent will mean they don't have new versions created for them

I think we need to use the ORM for this

Reply:
Here's the relevant code in the CMS that does this with the ORM for pages: https://github.com/silverstripe/silverstripe-cms/blob/4/code/Controllers/CMSMain.php#L817-L826

Reply:
Actually that's a bad example because it's doing exactly what you're doing here, but deliberately:

```php
                    // Nodes that weren't "actually moved" shouldn't be registered as
                    // having been edited; do a direct SQL update instead
```

We need to do the opposite for elements because changing the sort order is a versionable operation

Reply:
I disagree. If you have ~60 blocks and you move the last one to the beginning I don't think anyone would expect a new version on every single block. If any should have a new version, just the block that was reordered. 

Frankly, versioning sort is something that probably needs to be addressed separately. When you revert to a version with an old sort, the sort on other blocks will need to be changed.

We're doing this deliberately. Running 60 queries for updating sort is ridiculous when you can run two. Refactoring it to use `DBUpdate` is fair.

Reply:
>When you revert to a version with an old sort, the sort on other blocks will need to be changed.

That's a good point